src/prime_server.cpp

```diff
@@ -313,12 +312,16 @@ bool server_t<request_container_t, request_info_t>::enqueue(const zmq::message_t
   for (const auto& parsed_request : parsed_requests) {
     // figure out if we are expecting to close this request or not
     auto info = parsed_request.to_info(request_id++);
-
-    // if its enabled, see if its a health check
-    bool health_check = health_check_matcher && health_check_matcher(parsed_request);
+    
+    std::unique_ptr<zmq::message_t> shortcircuit;
+    if(shortcircuiter){
+      shortcircuit = shortcircuiter(parsed_request);
+    }else{
+      std::cout << "no shortcircuit" << std::endl;
+    }
```

Comment:
we could either do this
```suggestion
    auto shortcircuit = shortcircuiter ? shortcircuiter(parsed_request) : nullptr;
```

or in the constructor we can make the default shortcircuiter: `std::unique_ptr<zmq::message_t> [](const request_container_t&){return nullptr;}` instead of the empty one `{}`